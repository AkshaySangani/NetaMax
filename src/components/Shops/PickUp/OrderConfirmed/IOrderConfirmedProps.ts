import { IPickedUpOrderDetail } from "dataflows/Shops/DeliverOrder/IPickedUpOrderDetail";
import { IShopUser } from "dataflows/Shops/IShopUser";

export interface IOrderConfirmedProps {
  /**
   * On close panel action
   * @type {() => void}
   **/
  handleClose: () => void;

  /**
   * On copy action
   * @type {() => void}
   **/
  handleCopy: () => void;

  /**
   * The button text
   * @type {string}
   **/
  copyText: string;

  /**
   * The picked up order detail
   * @type {IPickedUpOrderDetail}
   **/
  orderDetail: IPickedUpOrderDetail | undefined;

  /**
   * The current shop selected
   * @type {IPickedUpOrderDetail}
   **/
  currentShop: IShopUser | undefined;
}
