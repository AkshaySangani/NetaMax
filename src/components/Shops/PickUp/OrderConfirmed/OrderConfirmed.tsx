import { Container, Flex, Box, DrawerBody, Button, DrawerFooter, Text } from "@chakra-ui/react";
import { DeliverButton } from "components/Shops/DeliverOrderButton/DeliverOrderButton.styled";
import { CURRENCY_NAME } from "constants/localeFormatConstants";
import React, { ReactElement } from "react";
import IconPickUpBasket from "styled/icons/Shops/PickUp/Basket";
import IconPickUpCheck from "styled/icons/Shops/PickUp/Check";
import { formatMoney } from "utils/currencyUtils";
import { splitHost } from "utils/stringUtils";
import { IOrderConfirmedProps } from "./IOrderConfirmedProps";

/**
 * The OrderConfirmed Component
 * @param {IOrderConfirmedProps} orderConfirmedProps The orderConfirmedProps props.
 * @returns {ReactElement} The OrderConfirmed component
 */
export const OrderConfirmed = (orderConfirmedProps: IOrderConfirmedProps): ReactElement => {
  const { handleCopy, copyText, orderDetail, handleClose, currentShop } = orderConfirmedProps;

  return (
    <>
      <Container background="#3CE88B" minHeight="40vh">
        <Flex direction="column" justifyContent="space-evenly" h="full">
          <Box>
            <IconPickUpCheck />
          </Box>
          <Box paddingBottom="10px">
            <Text fontSize="14px">Entrega confirmada</Text>
            <Text fontSize="32px" fontWeight="800" lineHeight="40px">
              Entrega confirmada correctamente
            </Text>
          </Box>
        </Flex>
      </Container>
      <DrawerBody paddingX="0px">
        <Container paddingX="16px" borderBottom="2px solid rgba(53, 53, 53, 0.1)" paddingY="24px">
          <Text fontSize="24px" fontWeight="800" color="#353535">
            Sigue compartiendo tu liga
          </Text>
          <Text fontWeight="400" fontSize="14px">
            Para que sigas ganando con tus entregas
          </Text>
          <Flex marginTop="16px" alignItems="center" justify="space-between">
            <Text fontSize="14px" fontWeight="600">
              {`https://neta.mx?${splitHost(currentShop?.Hosts || "")}`}
            </Text>
            <Button
              onClick={handleCopy}
              border="1px solid #353535"
              borderRadius="48px"
              paddingX="16px"
              paddingY="6px"
              background="white"
              minWidth="100px"
            >
              <Text fontSize="14px" fontWeight="600">
                {copyText}
              </Text>
            </Button>
          </Flex>
        </Container>
        <Container marginTop="18px" paddingX="16px">
          <Text>Resumen de compra</Text>
          <Flex marginBottom="24px">
            <Box mr="5px" height="18px">
              <IconPickUpBasket />
            </Box>
            <Text fontWeight="500" fontSize="14px" lineHeight="18px">
              {orderDetail?.productsAmount} Productos entregados
            </Text>
          </Flex>
          <Text fontSize="14px" fontWeight="400" color="#353535">
            Informaci√≥n de la orden
          </Text>
          <Flex marginTop="9px" alignItems="center" justify="space-between">
            <Text fontSize="14px" fontWeight="600">
              Subtotal
            </Text>
            <Text fontSize="16px" fontWeight="800">
              {formatMoney(Number(orderDetail?.orderSubtotalInclTax), CURRENCY_NAME)}
            </Text>
          </Flex>
          <Flex
            borderBottom="0.5px solid rgba(53, 53, 53, 0.2)"
            alignItems="center"
            justify="space-between"
            paddingBottom="8px"
          >
            <Text fontSize="14px" fontWeight="600">
              Descuento
            </Text>
            <Text fontSize="16px" fontWeight="800">
              {formatMoney(Number(orderDetail?.orderDiscount), CURRENCY_NAME)}
            </Text>
          </Flex>
          <Flex marginTop="9px" alignItems="center" justify="space-between">
            <Text fontSize="14px" fontWeight="600">
              Total
            </Text>
            <Text fontSize="16px" fontWeight="800">
              {formatMoney(Number(orderDetail?.orderTotal), CURRENCY_NAME)}
            </Text>
          </Flex>
        </Container>
      </DrawerBody>
      <DrawerFooter>
        <Box width="100%">
          <DeliverButton
            onClick={handleClose}
            height="48px"
            display="flex"
            alignItems="center"
            cursor="pointer"
          >
            <Text fontWeight="600">Regresar al inicio</Text>
          </DeliverButton>
        </Box>
      </DrawerFooter>
    </>
  );
};
