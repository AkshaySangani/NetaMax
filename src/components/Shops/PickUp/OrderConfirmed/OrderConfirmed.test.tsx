import { Drawer } from "@chakra-ui/react";
import { render, cleanup, screen, fireEvent } from "@testing-library/react";
import { OrderConfirmed } from "./OrderConfirmed";
import { IOrderConfirmedProps } from "./IOrderConfirmedProps";

describe("OrderConfirmed component", () => {
  let mock: IOrderConfirmedProps;
  const isOpen = true;
  const onClose = jest.fn();
  const handleCopy = jest.fn();
  const handleClose = jest.fn();

  /**
   * The OrderConfirmed Component
   * @returns {ReactElement} The OrderConfirmed component
   */
  const renderOrderConfirmed = () => {
    return (
      <Drawer
        isOpen={isOpen}
        placement="right"
        onClose={onClose}
        size="md"
        closeOnEsc
        closeOnOverlayClick
      >
        <OrderConfirmed {...mock} />
      </Drawer>
    );
  };

  beforeEach(() => {
    mock = {
      handleClose,
      handleCopy,
      copyText: "Algo a copiar",
      currentShop: {
        Hosts: "hosts-test",
        Id: "",
        CompanyName: "",
        Name: "",
        CompanyAddress: "",
        CompanyPhoneNumber: "",
      },
      orderDetail: {
        name: "",
        orderTotal: 0,
        orderSubtotalInclTax: 0,
        orderDiscount: 0,
        productsAmount: 0,
        productsOrder: [],
        customer: {
          id: "1063507",
          name: "asd",
          isNewUser: true,
          ZipPostalCode: "asd",
          username: "541134375450",
          email: "541134375450",
          billingAddress_Id: "86287",
          shippingAddress_Id: "",
          customerGuid: "027ce196-3b7c-409a-b33f-0e2966cf8c5f",
          isTaxExempt: true,
          affiliateId: "",
          vendorId: "",
          hasShoppingCartItems: false,
          deleted: false,
          registeredInStoreId: "",
          customerCurrentNetaWallet: "0.000000000000000",
          referralCode: "Milo-Molina-7620",
          firstName: "Milo Molina",
        },
      },
    };
  });

  afterEach(() => {
    cleanup();
  });

  test("OrderConfirmed should be render correctly", () => {
    const { container } = render(renderOrderConfirmed());
    expect(container).toMatchSnapshot();
  });

  test("Header should be render properly", () => {
    render(renderOrderConfirmed());
    expect(screen.getByText("Entrega confirmada correctamente")).toBeInTheDocument();
  });

  test("Body should be render properly", () => {
    render(renderOrderConfirmed());
    expect(screen.getByText("Sigue compartiendo tu liga")).toBeInTheDocument();
    expect(screen.getByText("Para que sigas ganando con tus entregas")).toBeInTheDocument();
    expect(screen.getByText("Subtotal")).toBeInTheDocument();
    expect(screen.getByText("Descuento")).toBeInTheDocument();
    expect(screen.getByText("Total")).toBeInTheDocument();
    expect(screen.getByText(mock.copyText)).toBeInTheDocument();
    fireEvent.click(screen.getByText(mock.copyText));
    expect(handleCopy).toHaveBeenCalled();
  });

  test("Footer should be render properly", () => {
    render(renderOrderConfirmed());
    expect(screen.getByText("Regresar al inicio")).toBeInTheDocument();
    fireEvent.click(screen.getByText("Regresar al inicio"));
    expect(handleClose).toHaveBeenCalled();
  });
});
