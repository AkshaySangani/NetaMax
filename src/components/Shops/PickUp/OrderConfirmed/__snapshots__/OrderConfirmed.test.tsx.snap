// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderConfirmed component OrderConfirmed should be render correctly 1`] = `<div />`;
